packages/db-objects-file/lib/objects/objectsInMemFileDB.js

```diff
@@ -750,8 +749,44 @@ class ObjectsInMemoryFileDB extends InMemoryFileDB {
         return keys.map(id => this.dataset[id]);
     }
 
+    objectEquivalent(a, b) {
+        if (a === null || a === undefined || b === null || b === undefined) {
+            return (a === b);
+        }
+        const aProps = Object.getOwnPropertyNames(a);
+        const bProps = Object.getOwnPropertyNames(b);
+
+        // different number of keys, different object
+        if (aProps.length !== bProps.length) {
+            return false;
+        }
+        // at least one key in the second object that is not in the first, different object
+        for (const key of bProps) {
+            if (!aProps.includes(key)) {
+                return false;
+            }
+        }
+
+        // we know keys are identical, so compare them
+        for (const key of aProps) {
+            // ignore ts and from
+            if (key === 'ts' || key === 'from') {
+                continue;
+            }
+            if (!isDeepStrictEqual(a[key], b[key])) {
+                return false;
+            }
+        }
+        return true;
+    }
```

Comment:
In general wouldn't it be more convenient to save `ts` and `from` of new obj to a constant, delete from both objects, perform a deep equal check and then add the ts and from of the new obj again? Not sure about performance, but should maybe be faster + much better readable

Reply:
Performance-wise, deleting keys from an object is pretty slow. Not sure if spreading into a new one and re-adding later is better. My bet is that the above approach is actually the fastest.

Reply:
Only Idea I had would only work in our veeeery special case (then I would kill the own method and icorporate it into the set method): set the "new" ts/from into the old object and then do a deepStrictEqual ... this ONLY works because we overwrite the "old" object afterwards anyway. What do you think about that?